cypress/integration/transforms_spec.js

```diff
@@ -0,0 +1,247 @@
+
+        /* Finds the correct header text, then navigates up the structure, then
+         * down to the next div, which contains the button. Then navigates down
+         * the div to the clickable button and clicks it.
+         */
+        cy.contains("taxless_total_price").parent().parent().parent().next()
+          .children().first().children().first().children().first()
+          .click({ force: true });
+        cy.contains("Aggregate by avg").click({ force: true });
+
+        // Confirm agg was added
+        cy.contains("avg_taxless_total_price");
+
+        // Click the next button
+        cy.get("button").contains("Next").click({ force: true });
+
+        // Confirm that we got to step 3 of creation page
+        cy.contains("Job enabled by default");
+
+        // Click the next button
+        cy.get("button").contains("Next").click({ force: true });
+
+        // Confirm that we got to step 4 of creation page
+        cy.contains("Review and create");
+
+        // Click the create button
+        cy.get("button").contains("Create").click({ force: true });
+
+        // Verify that sample data is add by checking toast notification
+        cy.contains(`Transform job "${TRANSFORM_ID}" successfully created.`);
+      });
+    });
+
+    describe("can be edited", () => {
+      beforeEach(() => {
+        cy.createTransform(TRANSFORM_ID, sampleTransform);
+        cy.reload();
+      });
+
+      it("successfully", () => {
+        // Confirm we have our initial transform
+        cy.contains(TRANSFORM_ID);
+
+        // Select checkbox for our transform
+        cy.get(`#_selection_column_${TRANSFORM_ID}-checkbox`)
+          .check({ force: true });
+
+        // Click on Actions popover menu
+        cy.get(`[data-test-subj="actionButton"]`).click({ force: true });
+
+        // Click Edit button
+        cy.get(`[data-test-subj="editButton"]`).click({ force: true });
+
+        // Wait for initial transform job to load
+        cy.contains("Test transform");
+
+        cy.get(`textArea[data-test-subj="description"]`).focus().clear().type("A new description");
+
+        // Click Save changes button
+        cy.get(`[data-test-subj="editTransformSaveButton"]`).click({ force: true });
+
+        // Confirm we get toaster saying changes saved
+        cy.contains(`Changes to transform saved`);
+
+        // Click into transform job details page
+        cy.get(`[data-test-subj="transformLink_${TRANSFORM_ID}"]`).click({ force: true });
+
+        // Confirm new description shows in details page
+        cy.contains("A new description");
+      });
+    });
+
+    describe("can be deleted", () => {
+      beforeEach(() => {
+        cy.createTransform(TRANSFORM_ID, sampleTransform);
+        cy.reload();
+      });
+
+      it("successfully", () => {
+        // Confirm we have our initial transform
+        cy.contains(TRANSFORM_ID);
+
+        // Disable transform
```

Comment:
Does a transform need to be disabled to be deleted?

Reply:
Yes, an exception is thrown otherwise.